server/stream.go

```diff
@@ -1055,6 +1090,11 @@ func (mset *stream) update(config *StreamConfig) error {
 	}
 	mset.mu.Unlock()
 
+	if js != nil {
+		js.releaseStreamResources(&ocfg)
+		js.reserveStreamResources(cfg)
```

Comment:
In addition, the same diff is needed here.
diff := oldMaxBytes - newMaxBytes 
diff > 0 -> reserve diff
diff == 0 -> do nothing
diff < 0 -> release by -diff